builders/integrations/oracles/supra.md

```diff
@@ -0,0 +1,237 @@
+---
+title: Supra Oracles
+description: Supra's Pull Oracle provides low-latency, on-demand price feed updates for a variety of use cases. Learn how to integrate Supra's oracle on Moonbeam.
+---
+
+# Supra Oracles
+
+## Introduction {: #introduction }
+
+[Supra](https://supraoracles.com){target=\_blank} is a novel, high-throughput oracle and intralayer: a vertically integrated toolkit of cross-chain solutions (data oracles, asset bridges, automation network, and more) that interlink all blockchains, public (L1s and L2s) or private (enterprises), including Moonbeam.
+
+Supra provides decentralized oracle price feeds that can be used for on-chain and off-chain use cases such as spot and perpetual DEXes, lending protocols, and payment protocols.
+
+In this guide, you'll learn how to use Supra Oracles' price feeds to fetch price data in smart contracts on Moonbeam.
+
+--8<-- 'text/_disclaimers/third-party-content-intro.md'
+
+## How to use Supra's Price Feeds {: #price-feeds }
+
+Supra uses a pull model as a customized approach that publishes price data upon request. It combines Web2 and Web3 methods to achieve low latency when sending data from Supra to destination chains. The process involves the following steps:
+
+1. Web2 methods are used to retrieve price data from Supra
+2. Smart contracts are utilized for cryptographically verifying and writing the latest price data on-chain, where it lives on immutable ledgers, using [Supra's Pull Oracle V1](https://supra.com/docs/data-feeds/pull-model){target=\_blank}
+3. Once the data has been written on-chain, the most recently published price feed data will be available in Supra's Storage contract
+
+You can fetch price data from Supra for any [available data pairs](https://supra.com/docs/data-feeds/data-feeds-index/){target=\_blank}.
+
+The Pull Oracle contract on Moonbeam is located at the following address:
+
+=== "Moonbeam"
+
+    ```text
+    {{ networks.moonbeam.supra.pull_oracle }}
+    ```
+
+=== "Moonbase Alpha"
```

Comment:
I noticed that the contracts are verified for Moonbeam but not for Moonbase Alpha. Is it possible for the Supra team to verify these for Moonbase Alpha as well?